server/src/routes/guards.ts

```diff
@@ -21,11 +23,53 @@ export type RouterContext = Router.RouterContext<
 >;
 
 const basicAuthAdmin = auth({ name: config.admin.username, pass: config.admin.password });
+
 export const basicAuthAws = auth({
   name: config.users.cloud.username,
   pass: config.users.cloud.password,
 });
 
+export const crossCheckGuard = async (ctx: RouterContext, next: () => Promise<void>) => {
+  const user = ctx.state.user;
+  ctx.params.courseId = Number(ctx.params.courseId);
+
+  if (user) {
+    const guards = userGuards(user);
+    const { courseId } = ctx.params;
+
+    if (guards.isLoggedIn(ctx) && guards.isPowerUser(courseId)) {
+      await next();
+      return;
+    }
+
+    setErrorResponse(ctx, StatusCodes.FORBIDDEN, ReasonPhrases.FORBIDDEN);
+  } else {
+    const authHeader = ctx.request.header.authorization as string | undefined;
```

Comment:
umm.. I think you over engineered it a little bit. You just need to do the following and it will be fine. admin was mostly for debugging purposes and I don't think we need to all this code to support it.

```
await basicAuthAws(ctx, next);
```